components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/AdminForcedPasswordResetHandler.java

```diff
@@ -72,20 +75,102 @@ private void handleUpdateCredentialsByAdmin(Map<String, Object> eventProperties,
             throws IdentityEventException {
 
         User user = getUser(eventProperties, userStoreManager);
+        String maskedUsername = Utils.maskIfRequired(user.getUserName());
+
         if (log.isDebugEnabled()) {
             log.debug("PostUpdateCredentialsByAdmin - AdminForcedPasswordResetHandler for user : "
-                    + user.toString());
+                    + maskedUsername);
         }
 
         UserRecoveryData userRecoveryData = getRecoveryData(user);
         if (userRecoveryData != null) {
             invalidateRecoveryData(user);
             if (log.isDebugEnabled()) {
                 log.debug("PostUpdateCredentialsByAdmin - invalidate Recovery Data for user : "
-                        + user.toString());
+                        + maskedUsername);
+            }
+        }
+
+        handleAccountUnlockOnAdminPasswordReset(user, userStoreManager);
+    }
+
+    /**
+     * Handles the account unlock process after an admin password reset.
+     *
+     * @param user             User object.
+     * @param userStoreManager UserStoreManager instance.
+     * @throws IdentityEventException If an error occurs while handling account unlock.
+     */
+    private void handleAccountUnlockOnAdminPasswordReset(User user, UserStoreManager userStoreManager)
+            throws IdentityEventException {
+
+        try {
+            // Retrieve the user's account lock-related claims (locked status, locked reason, and account state).
+            Map<String, String> currentClaims = userStoreManager.getUserClaimValues(
+                    user.getUserName(),
+                    new String[]{
+                            IdentityRecoveryConstants.ACCOUNT_LOCKED_CLAIM,
+                            IdentityRecoveryConstants.ACCOUNT_LOCKED_REASON_CLAIM,
+                            IdentityRecoveryConstants.ACCOUNT_STATE_CLAIM_URI
+                    },
+                    null
+            );
+
+            if (currentClaims == null) {
+                return;
+            }
+
+            boolean isAccountLocked =
+                    Boolean.parseBoolean(currentClaims.get(IdentityRecoveryConstants.ACCOUNT_LOCKED_CLAIM));
+            String accountState = currentClaims.get(IdentityRecoveryConstants.ACCOUNT_STATE_CLAIM_URI);
+            String lockedReason = currentClaims.get(IdentityRecoveryConstants.ACCOUNT_LOCKED_REASON_CLAIM);
+
+            // Prepare a map for claim updates.
+            Map<String, String> claimMap = new HashMap<>();
+
+            if (isAccountLocked) {
+                // Scenario 1:
+                // If the account is locked and the locked reason is valid and qualifies for unlocking criteria
+                // (pending password update), proceed with unlocking.
+                if (isUnlockRequired(lockedReason)) {
```

Comment:
```suggestion
                if (isAccountUnlockRequiredAfterForcedPasswordReset(lockedReason)) {
```

Reply:
We can further improve as below.
if (!isAccountUnlockRequiredAfterForcedPasswordReset(lockedReason)) {
   return;
}

.
.
.
// Not require this check.   if (!claimMap.isEmpty()) {}

Reply:
Resolved via https://github.com/wso2-extensions/identity-governance/pull/936/commits/84795f158884a56df8f24b2b549b0f3ebebc3681.

> Change the implementation.
> 
> The method name and its comment should indicate that account should not get unlocked for lock reasons except those reasons.

Had an offline discussion and agreed to proceed with the existing approach.